zerver/lib/message.py

```diff
@@ -235,6 +237,15 @@ def re_render_content_for_management_command(message):
     message.rendered_content_version = bugdown.version
     message.save_rendered_content()
 
+def get_mentions_by_message_id(message_id):
+    # type: (int) -> List[str]
+    mentions = []
+    ums = UserMessage.objects.filter(message=message_id,
+                                     flags=UserMessage.flags.mentioned)
+    for um in ums:
+        mentions.append(um.user_profile.full_name)
+    return mentions
```

Comment:
While this code works, it will create a performance issue, because it is adding additional database queries for each of the 1000 messages fetched when loading the homepage!  I'll look into how to solve this correctly.